### ISSUE TITLE
python 3.13 support

### ISSUE THREAD

**Issue by eymenbilge** [IfcOpenShell/IfcOpenShell#5566]
Hi,

when trying to install on python 3.13 it gives error

C:\Users\ecbil>pip install ifcopenshell
ERROR: Ignored the following versions that require a different python version: 0.7.0.240406 Requires-Python <3.13,>=3.9; 0.7.0.240418 Requires-Python <3.13,>=3.9; 0.7.0.240423 Requires-Python <3.13,>=3.9; 0.7.0.240518 Requires-Python <3.13,>=3.9; 0.7.0.240521 Requires-Python <3.13,>=3.9; 0.7.0.240627 Requires-Python <3.13,>=3.9; 0.7.10 Requires-Python <3.13,>=3.9; 0.8.0 Requires-Python <3.13,>=3.9
ERROR: Could not find a version that satisfies the requirement ifcopenshell (from versions: none)
ERROR: No matching distribution found for ifcopenshell

**Comment by francianeax4b** [comment]
Here we have the same error. Could anyone help?

**Comment by aothms** [comment]
There are builds for 3.13 now, but it will take some time before these are pip-installable. You can manually download the zips and extract them into your site-packages folder in the meantime https://github.com/IfcOpenBot/IfcOpenShell/commit/92b63a0ef24455e89836d02b4c9bb011b0006f2e

**Comment by faridcher** [comment]
please also add python 3.13  support to Bonsai. Too often, IFCopenShell (and Bonsai) falls behind many linux distributions Blender and Python upgrades. E.g Debian Sid Blender depends on python3.13.* now.
https://packages.debian.org/sid/blender
Arch linux as well:
https://archlinux.org/packages/extra/x86_64/blender/

blender error
```
Extension bl_ext.user_default.bonsai is incompatible (This Python version (3.13) isn't compatible with (3.12))
```

**Comment by ShuddiCheckAI** [comment]
In the requirement txt what  bliss is compatible for python 3.13 ?

**Comment by sanzoghenzo** [comment]
Hi, why was this closed? version 0.8.2 still doesn't have python 3.13 packages in the release page (and on pypi)

**Comment by autra** [comment]
Indeed, python 3.13 is still not supported. From my (incomplete) research, we need python 3.13 support for mathutils as a prerequisite. I think this issue should be reopened.

**Comment by Moult** [comment]
Yeah it's an omission in the build system, reopening. Bonsai won't have 3.13 support until Blender does, but IfcOpenShell can.

**Comment by faridcher** [comment]
To workaround this issue, I use the official Blender distribution rather than my Linux distro version of Blender. It comes with a built-in python 3.11 and Bonsai with python 3.11 can be painlessly installed and used.

[Blender Download Link](https://mirror.freedif.org/blender/release/Blender4.4/blender-4.4.3-linux-x64.tar.xz)
[Bonsai Link](https://github.com/IfcOpenShell/IfcOpenShell/releases/tag/bonsai-0.8.4-alpha2507162053)

**Comment by autra** [comment]
Prerequisite: https://gitlab.com/ideasman42/blender-mathutils/-/issues/36

**Comment by sanzoghenzo** [comment]
> To workaround this issue, I use the official Blender distribution rather than my Linux distro version of Blender. It comes with a built-in python 3.11 and Bonsai with python 3.11 can be painlessly installed and used.
> 
> [Blender Download Link](https://mirror.freedif.org/blender/release/Blender4.4/blender-4.4.3-linux-x64.tar.xz) [Bonsai Link](https://github.com/IfcOpenShell/IfcOpenShell/releases/tag/bonsai-0.8.4-alpha2507162053)

If you only need the ifcopenshell library, you can create a virtual environment with a specific python version using pyenv, or my favourite, uv:

```shell
uv init --python 3.11 my-project
cd my-project
uv add ifcopenshell
```

**Comment by Andrej730** [comment]
Added Python 3.13 build to pypi.

<img width="674" height="324" alt="Image" src="https://github.com/user-attachments/assets/f5079192-d6f4-458c-b036-be79a883716a" />

---

@autra 

We don't use mathutils anymore, so it's not really important to us.

Anyways, Blender seems to gave up maintaining `mathutils` as standalone package 3 years ago:

<img width="497" height="176" alt="Image" src="https://github.com/user-attachments/assets/cef48d19-e89e-4ab7-89cb-946cd7da64da" />

And the only way to access `mathutils` outside Blender is installing https://pypi.org/project/bpy and importing it:
```
.venv\scripts\python.exe -c "import bpy; import mathutils; print(mathutils)"
```

So it's likely that your issue will remain unresolved. Though you can try to contact devs on https://chat.blender.org/, if you're still using `mathutils` for your own scripts and want to help Blender updating `mathutils` on pypi.

**Comment by autra** [comment]
Thanks @Andrej730, no I don't need mathutils on its own, so that's perfect. I plan on updating the nixpkgs package soon.

**Comment by mtekman** [comment]
Hi there, thanks for making a 3.13 build of ifcopenshell so quickly

I'm trying to install bonsai on blender 4.5.0 which seems to be using my system library of python 3.13.5

Does this build of ifcopenshell p313 coincide with a bonsai build? 

Apologies if they are completely unrelated issues

**Comment by khameeteman** [comment]
Hi thanks for adding support for Python 3.13.

I am running into an issue while installing on Linux, anyone experiencing the same?

<img width="1912" height="604" alt="Image" src="https://github.com/user-attachments/assets/c5541eed-fd45-44d1-a7ee-4070f28aecde" />

EDIT: moved here https://github.com/IfcOpenShell/IfcOpenShell/issues/6989

**Comment by Andrej730** [comment]
@khameeteman please start a separate issue
